<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>680</width>
    <height>520</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MAZE GENERATOR</string>
  </property>
  <property name="styleSheet">
   <string notr="true">  
MainWindow {
    background-color: #f0f0f0;
}

QLabel {
    font-family: 'Segoe UI', Arial, sans-serif;
    font-size: 12px;
    color: #333333;
}

QPushButton {
    font-family: 'Segoe UI', Arial, sans-serif;
    font-size: 12px;
    padding: 5px 10px;
    border: 1px solid #007bff;
    border-radius: 5px;
    background-color: #007bff;
    color: #ffffff;
}

QPushButton:hover {
    background-color: #0056b3;
}

QLineEdit, QSpinBox {
    font-family: 'Segoe UI', Arial, sans-serif;
    font-size: 12px;
    padding: 5px;
    border: 1px solid #dcdcdc;
    border-radius: 5px;
}

QGridLayout {
    spacing: 10px;
}

MazeView {
    border: none;
    background-color: white;
}
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="MazeView" name="mazeView">
    <property name="geometry">
     <rect>
      <x>5</x>
      <y>5</y>
      <width>510</width>
      <height>510</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
   </widget>
   <widget class="QFrame" name="selectFileQFrame">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>5</y>
      <width>154</width>
      <height>68</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QFrame {
    background-color: #ffffff;
    border: 1px solid #dcdcdc;
    border-radius: 5px;
}</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLineEdit" name="selectFileQLineEdit">
     <property name="geometry">
      <rect>
       <x>14</x>
       <y>6</y>
       <width>128</width>
       <height>25</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="selectFileQPushButton">
     <property name="geometry">
      <rect>
       <x>27</x>
       <y>39</y>
       <width>102</width>
       <height>22</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Segoe UI</family>
       <pointsize>-1</pointsize>
       <weight>50</weight>
       <bold>false</bold>
      </font>
     </property>
     <property name="text">
      <string>SELECT FILE</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="findPathQFrame">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>78</y>
      <width>155</width>
      <height>124</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QFrame#findPathQFrame {
    background-color: #ffffff;
    border: 1px solid #dcdcdc;
    border-radius: 5px;
}</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QWidget" name="horizontalLayoutWidget_3">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>30</y>
       <width>143</width>
       <height>33</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="startCordQBoxLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="label">
          <property name="maximumSize">
           <size>
            <width>20</width>
            <height>35</height>
           </size>
          </property>
          <property name="font">
           <font>
            <family>Segoe UI</family>
            <pointsize>-1</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>X</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="xFromCordQSpinBox">
          <property name="font">
           <font>
            <family>Segoe UI</family>
            <pointsize>-1</pointsize>
            <weight>50</weight>
            <bold>false</bold>
           </font>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="keyboardTracking">
           <bool>true</bool>
          </property>
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>0</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="maximumSize">
           <size>
            <width>20</width>
            <height>35</height>
           </size>
          </property>
          <property name="font">
           <font>
            <family>Segoe UI</family>
            <pointsize>-1</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Y</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="yFromCordQSpinBox">
          <property name="font">
           <font>
            <family>Segoe UI</family>
            <pointsize>-1</pointsize>
            <weight>50</weight>
            <bold>false</bold>
           </font>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="keyboardTracking">
           <bool>true</bool>
          </property>
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>0</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>6</y>
       <width>155</width>
       <height>26</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Segoe UI</family>
       <pointsize>-1</pointsize>
      </font>
     </property>
     <property name="text">
      <string>― START POINT ―</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QWidget" name="horizontalLayoutWidget_4">
     <property name="geometry">
      <rect>
       <x>6</x>
       <y>87</y>
       <width>142</width>
       <height>33</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="finishCordQBoxLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="maximumSize">
           <size>
            <width>20</width>
            <height>35</height>
           </size>
          </property>
          <property name="font">
           <font>
            <family>Segoe UI</family>
            <pointsize>-1</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>X</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="xToCordQSpinBox">
          <property name="font">
           <font>
            <family>Segoe UI</family>
            <pointsize>-1</pointsize>
            <weight>50</weight>
            <bold>false</bold>
           </font>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="keyboardTracking">
           <bool>true</bool>
          </property>
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>0</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="label_4">
          <property name="maximumSize">
           <size>
            <width>20</width>
            <height>35</height>
           </size>
          </property>
          <property name="font">
           <font>
            <family>Segoe UI</family>
            <pointsize>-1</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Y</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="yToCordQSpinBox">
          <property name="font">
           <font>
            <family>Segoe UI</family>
            <pointsize>-1</pointsize>
            <weight>50</weight>
            <bold>false</bold>
           </font>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="keyboardTracking">
           <bool>true</bool>
          </property>
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>0</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>60</y>
       <width>152</width>
       <height>29</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Segoe UI</family>
       <pointsize>-1</pointsize>
      </font>
     </property>
     <property name="text">
      <string>― FINISH POINT ―</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="generateMazeQFrame">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>207</y>
      <width>156</width>
      <height>69</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QFrame {
    background-color: #ffffff;
    border: 1px solid #dcdcdc;
    border-radius: 5px;
}</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QPushButton" name="generateMazeQPushButton">
     <property name="geometry">
      <rect>
       <x>14</x>
       <y>40</y>
       <width>129</width>
       <height>22</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Segoe UI</family>
       <pointsize>-1</pointsize>
      </font>
     </property>
     <property name="text">
      <string>GENERATE MAZE</string>
     </property>
    </widget>
    <widget class="QWidget" name="gridLayoutWidget">
     <property name="geometry">
      <rect>
       <x>27</x>
       <y>4</y>
       <width>106</width>
       <height>31</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="mazeGenerateQGridLayout">
      <item row="0" column="0">
       <widget class="QSpinBox" name="rowsQSpinBox">
        <property name="font">
         <font>
          <family>Segoe UI</family>
          <pointsize>-1</pointsize>
          <weight>50</weight>
          <bold>false</bold>
         </font>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="keyboardTracking">
         <bool>true</bool>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>50</number>
        </property>
        <property name="value">
         <number>10</number>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="columnsQSpinBox">
        <property name="font">
         <font>
          <family>Segoe UI</family>
          <pointsize>-1</pointsize>
          <weight>50</weight>
          <bold>false</bold>
         </font>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="keyboardTracking">
         <bool>true</bool>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>50</number>
        </property>
        <property name="value">
         <number>10</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>521</x>
      <y>282</y>
      <width>155</width>
      <height>39</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QFrame {
    background-color: #ffffff;
    border: 1px solid #dcdcdc;
    border-radius: 5px;
}</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QPushButton" name="caveSelectQPushButton">
     <property name="geometry">
      <rect>
       <x>6</x>
       <y>8</y>
       <width>142</width>
       <height>23</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Segoe UI</family>
       <pointsize>-1</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>CAVE </string>
     </property>
    </widget>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MazeView</class>
   <extends>QGraphicsView</extends>
   <header>mazeview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
